src/pages/home/profile/Profile.tsx

```diff
@@ -1,17 +1,213 @@
+import * as Yup from "yup";
 
 const Profile: FunctionComponent = () => {
+  const dispatch = useDispatch();
+  const { roles, genres } = useSelector(selectContent);
+  const {
+    profile: {
+      email,
+      firstName,
+      genre,
+      lastName,
+      nickname,
+      pictureUrl,
+      role,
+    } = {},
+  } = useSelector(selectSession);
+
+  const initialValues = {
+    firstName,
+    lastName,
+    email,
+    nickname,
+    role,
+    genre,
+    currentPassword: "",
+    newPassword: "",
+    confirmPassword: "",
+  };
+
+  const validationSchema = Yup.object({
+    firstName: commonYupValidation.firstName,
+    lastName: commonYupValidation.lastName,
+    email: commonYupValidation.email,
+    nickname: commonYupValidation.nickname,
+    role: commonYupValidation.role(roles),
+    genre: commonYupValidation.role(genres),
+    currentPassword: Yup.string().when("newPassword", {
+      is: (currentValue: string) => currentValue,
+      then: Yup.string().required("Current password is required"),
+    }),
+    newPassword: commonYupValidation.newPassword,
+    confirmPassword: commonYupValidation.confirmPassword.when("newPassword", {
+      is: (currentValue: string) => currentValue,
+      then: Yup.string().required("Confirm new password is required"),
+    }),
+  });
+
+  /**
+   * Update profile data with modifications made.
+   */
+  const handleSubmit = (values: FormikValues) => {
+    dispatch(updateProfile({ ...values }));
+  };
+
   return (
-    <Box
-      display="flex"
-      flexGrow={ 1 }
-      justifyContent="center"
-      alignItems="center"
+    <Container
+      maxWidth={ false }
+      sx={ {
+        marginLeft: [null, null, 4.5],
+        overflow: "auto",
+        paddingY: 7.5,
+        textAlign: ["center", "center", "initial"],
+      } }
     >
-      <Typography>Profile page</Typography>
-    </Box>
+      <Typography variant="h1" mb={ 5 }>
+        YOUR PROFILE
+      </Typography>
+      { pictureUrl && (
+        <ProfileImage
+          alt="Profile picture"
+          src={ pictureUrl }
+          sx={ { marginBottom: 5 } }
+        />
+      ) }
+      <Formik
+        initialValues={ initialValues }
+        onSubmit={ handleSubmit }
+        validationSchema={ validationSchema }
+      >
+        { ({
+          isValid,
+          values: { currentPassword, newPassword, confirmPassword },
+        }) => {
+          const showEndAdornment = !!(
+            currentPassword ||
+            newPassword ||
+            confirmPassword
+          );
+          return (
+            <Form>
+              <Stack
+                sx={ {
+                  display: "grid",
+                  gridTemplateColumns: [
+                    "repeat(1, 1fr)",
+                    null,
+                    "repeat(2, 1fr)",
+                  ],
+                  columnGap: [undefined, undefined, "20px"],
+                  maxWidth: [undefined, undefined, "700px"],
```

Comment:
interesting find here: you'd think that null and undefined would give you similar results.... they do not.

null will insert a value that fits the media query of the screen and still show up... while undefined will not show a selector/property at all.